service/microcloud.go

```diff
@@ -66,44 +62,7 @@ func NewCloudService(name string, addr string, dir string) (*CloudService, error
 }
 
 // StartCloud launches the MicroCloud daemon with the appropriate hooks.
-func (s *CloudService) StartCloud(ctx context.Context, service *Handler, endpoints []rest.Endpoint, verbose bool, debug bool, heartbeatInterval time.Duration) error {
-	args := microcluster.DaemonArgs{
-		Verbose:           verbose,
-		Debug:             debug,
-		Version:           version.RawVersion,
-		HeartbeatInterval: heartbeatInterval,
-
-		PreInitListenAddress: "[::]:" + strconv.FormatInt(CloudPort, 10),
-		Hooks: &state.Hooks{
-			PostJoin: func(ctx context.Context, s state.State, cfg map[string]string) error {
-				// If the node has joined close the session.
-				// This will signal to the client to exit out gracefully
-				// and ultimately lead to the closing of the websocket connection.
-				// Prevent blocking of the hook by also watching the outer context.
-				select {
-				case service.Session.ExitCh() <- true:
-				case <-ctx.Done():
-				}
-
-				return nil
-			},
-			OnStart: service.Start,
-		},
-		ExtensionServers: map[string]rest.Server{
-			"microcloud": {
-				CoreAPI:   true,
-				PreInit:   true,
-				ServeUnix: true,
-				Resources: []rest.Resources{
-					{
-						PathPrefix: types.APIVersion,
-						Endpoints:  endpoints,
-					},
-				},
-			},
-		},
-	}
-
+func (s *CloudService) StartCloud(ctx context.Context, args microcluster.DaemonArgs) error {
 	return s.client.Start(ctx, args)
```

Comment:
Mhh, in case the daemon gets restarted neither the `PostBootstrap` nor `PostJoin` hooks are getting called. 
Should we set the address here after starting the microcluster daemon?

Reply:
Good point. I've moved the `Start` hook out into the daemon CLI package and now it also updates the handler address if the database is already open.

Reply:
Had to make a small adjustment because `Database.IsOpen` returns "valid" errors in the case of upgrades.

Reply:
Looks good! Probably StartCloud func can be removed now too as microcluster's Start can be called right away.